corehq/apps/userreports/tests/test_pillow.py

```diff
@@ -251,36 +251,109 @@ def test_process_doc_from_sql_stale(self):
+            form, cases = post_case_blocks(
+                [
+                    CaseBlock(
+                        create=i == 0,
+                        case_id='parent-id',
+                        case_name='parent-name',
+                        case_type='bug',
+                        update={'update-prop-parent': i},
+                    ).as_xml(),
+                    CaseBlock(
+                        create=i == 0,
+                        case_id='child-id',
+                        case_name='child-name',
+                        case_type='bug-child',
+                        index={'parent': ('bug', 'parent-id')},
+                        update={'update-prop-child': i}
+                    ).as_xml()
+                ], domain=self.domain
+            )
+            indicators = AsyncIndicator.objects.filter(doc_id='child-id', pillow=self.pillow.pillow_id)
+            self.assertEqual(indicators.count(), 0)
+
+            self.pillow.process_changes(since=since, forever=False)
+            self.assertEqual(indicators.count(), 1)
+
+            queue_async_indicators()
+            rows = self.adapter.get_query_object()
+            self.assertEqual(rows.count(), 1)
+
+            row = rows[0]
+            self.assertEqual(int(row.parent_property), i)
+
+            errors = PillowError.objects.filter(doc_id='child-id', pillow=self.pillow.pillow_id)
+            self.assertEqual(errors.count(), 0)
+            self.assertEqual(indicators.count(), 0)
+
+    @patch('corehq.apps.userreports.tasks._get_config_by_id')
+    def test_async_save_fails(self, config):
+        config.return_value = None
+        since = self.pillow.get_change_feed().get_latest_offsets()
+        form, cases = post_case_blocks(
+            [
+                CaseBlock(
+                    create=True,
+                    case_id='parent-id',
+                    case_name='parent-name',
+                    case_type='bug',
+                    update={'update-prop-parent': 0},
+                ).as_xml(),
+                CaseBlock(
+                    create=True,
+                    case_id='child-id',
+                    case_name='child-name',
+                    case_type='bug-child',
+                    index={'parent': ('bug', 'parent-id')},
+                    update={'update-prop-child': 0}
+                ).as_xml()
+            ], domain=self.domain
+        )
+
+        indicators = AsyncIndicator.objects.filter(doc_id='child-id', pillow=self.pillow.pillow_id)
+        self.assertEqual(indicators.count(), 0)
+        self.pillow.process_changes(since=since, forever=False)
+        self.assertEqual(indicators.count(), 1)
+
+        queue_async_indicators()
+        rows = self.adapter.get_query_object()
+        self.assertEqual(rows.count(), 0)
+
+        errors = PillowError.objects.filter(doc_id='child-id', pillow=self.pillow.pillow_id)
+        self.assertEqual(errors.count(), 0)
```

Comment:
Since there are multiple assertions in the test, a small explanatory comment wherever applicable is nice to have